plugin/ui/completion.py

```diff
@@ -323,45 +324,38 @@ class _PhantomCompletion(_BaseCompletion):
     def __init__(self, view: sublime.View) -> None:
         super().__init__(view)
 
-        self._phantom_set = self._get_phantom_set(view, True)
+        self._phantom_set = self._get_phantom_set(view)
 
     @classmethod
-    def _get_phantom_set(cls, view: sublime.View, init: bool = False) -> Optional[sublime.PhantomSet]:
+    def _get_phantom_set(cls, view: sublime.View) -> sublime.PhantomSet:
         view_id = view.id()
-        phantom_set = _phantom_sets_per_view.get(view_id, None)
 
-        if init:
-            phantom_set = sublime.PhantomSet(view, cls.COPILOT_PHANTOM_COMPLETION)
-            _phantom_sets_per_view[view_id] = phantom_set
+        # create phantom set if there is no existing one
+        if not _view_to_phantom_set.get(view_id):
+            _view_to_phantom_set[view_id] = sublime.PhantomSet(view, cls.COPILOT_PHANTOM_COMPLETION)
 
-        return phantom_set
+        return _view_to_phantom_set[view_id]
 
     def normalize_phantom_line(self, line: str) -> str:
-        # return re.sub(r"( {2,})", lambda m: "&nbsp;" * len(m.group(0)), html.escape(line))
-        return (
-            html.escape(line).replace(" ", "&nbsp;").replace("\t", "&nbsp;" * self.view.settings().get("tab_size", 4))
-        )
+        return html.escape(line).replace(" ", "&nbsp;").replace("\t", "&nbsp;" * self.view.settings().get("tab_size"))
 
-    def _build_phantom(
-        self, lines: Union[str, List[str]], begin: int, end: Optional[int] = None, inline: bool = True
-    ) -> sublime.Phantom:
-        body = (
-            "".join(
-                self.PHANTOM_LINE_TEMPLATE.format(
-                    class_name=("first" if index == 0 else ""), content=self.normalize_phantom_line(line)
-                )
-                for index, line in enumerate(lines)
+    def _build_phantom(self, lines: Sequence[str], region: sublime.Region, *, inline: bool = True) -> sublime.Phantom:
+        view_settings = self.view.settings()
+
+        body = "".join(
+            self.PHANTOM_LINE_TEMPLATE.format(
+                class_name="rest" if index else "first",
+                content=self.normalize_phantom_line(line),
             )
-            if isinstance(lines, list)
-            else self.normalize_phantom_line(lines)
+            for index, line in enumerate(lines)
```

Comment:
With this refactoring, `copilot-completion-line` class will be added even for the first inline phantom, but I wanted to avoid this because it adds an extra margin.

Reply:
Oops, didn't notice this. I think there is no good common way to refactor this then. Will revert.